crates/wit-component/tests/interfaces/empty/types_only.wat

```diff
@@ -1,10 +1 @@
-(component
-  (type (;0;)
-    (instance)
-  )
-  (type (;1;)
-    (instance)
-  )
-  (import "empty" (instance (;0;) (type 0)))
-  (export "empty" (type 1))
-)
\ No newline at end of file
+(component)
```

Comment:
I agree with you. This is an artifact of the fuzzer I wrote where the types-only document and the component-based document are required to be equal, but instead I think it would be better to go for a lossless representation of a document as a binary component. In https://github.com/WebAssembly/component-model/pull/141 I'm proposing a much different binary format than is currently emitted here but should be suitable for a lossless representation of a document.

Additionally my plan currently is to remove `--types-only` and instead have:

* `wasm-tools component wit component.wasm` - prints the WIT interface of an existing component
* `wasm-tools component wit foo.wit` - prints the "single string" version of a WIT document after resolving all local filesystem paths and such
* `wasm-tools component wit foo.wit --binary` - prints the binary format as specified in https://github.com/WebAssembly/component-model/pull/141 (with a standard `-t` flag for using the wasm text format)
* `wasm-tools component wit wit.wasm` - prints the WIT document for a binary-encoding wit file

Basically moving the "types only" mode to `wasm-tools component wit --binary <input-doc>` instead of part of `wasm-tools component new`. Bikeshedding is of course welcome on all this!

Reply:
Oh, I completely missed that there was an alternative encoding scheme proposed in that PR. Let me review the upstream PR to get up-to-speed, but :+1: on not being able to round-trip certain constructs until the encoding is updated.

Reply:
Ah yeah this space has been moving a bit quickly and basically changed the moment I posted my own PR here. I plan to talk more about what's going on at the component-model meeting tomorrow.